scalastyle-config.xml

```diff
@@ -0,0 +1,79 @@
+<scalastyle>
+  <name>Scalastyle standard configuration</name>
+
+  <check level="error" enabled="true" class="org.scalastyle.file.FileTabChecker"/>
+  <check level="error" enabled="true" class="org.scalastyle.file.WhitespaceEndOfLineChecker"/>
+  <check level="error" enabled="true" class="org.scalastyle.file.NewLineAtEofChecker"/>
+
+  <!--Indentation checker, whilst useful gives false positives on the file headers, so disabled for now-->
+  <check level="error" enabled="false" class="org.scalastyle.file.IndentationChecker"/>
+
+  <check level="error" enabled="true" class="org.scalastyle.file.FileLineLengthChecker">
+    <parameters>
+      <parameter name="maxLineLength"><![CDATA[120]]></parameter>
+    </parameters>
+  </check>
+
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.RedundantIfChecker"/>
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.SimplifyBooleanExpressionChecker"/>
+
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.UppercaseLChecker"/>
+
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.ClassNamesChecker">
+    <parameters>
+      <parameter name="regex"><![CDATA[[A-Z][A-Za-z]*]]></parameter>
+    </parameters>
+  </check>
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.ObjectNamesChecker">
+    <parameters>
+      <parameter name="regex"><![CDATA[[A-Z][A-Za-z]*]]></parameter>
+    </parameters>
+  </check>
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.PackageObjectNamesChecker">
+    <parameters>
+      <parameter name="regex"><![CDATA[^[a-z][A-Za-z]*$]]></parameter>
+    </parameters>
+  </check>
+
+  <check level="error" enabled="false" class="org.scalastyle.scalariform.EqualsHashCodeChecker"/>
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.CovariantEqualsChecker"/>
+
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.IllegalImportsChecker">
+    <parameters>
+      <parameter name="illegalImports"><![CDATA[sun._,java.awt._]]></parameter>
+    </parameters>
+  </check>
+
+  <check level="error" enabled="true" class="org.scalastyle.scalariform.ReturnChecker"/>
+
+  <check level="error" enabled="false" class="org.scalastyle.scalariform.IfBraceChecker">
```

Comment:
Disabled checkers are those I haven't tried yet, but whom I intend to apply or at least try at some point. Checkers I clearly disagreed with have been removed entirely.

I'll put that as a comment in the beginning of the file.